client/dive-common/store/context.ts

```diff
@@ -1,26 +1,43 @@
 import Install, { reactive } from '@vue/composition-api';
-import Vue from 'vue';
+import Vue, { VueConstructor } from 'vue';
 /* Components */
 import TypeThreshold from 'dive-common/components/TypeThreshold.vue';
 
 Vue.use(Install);
 
-
-const componentMap = {
-  TypeThreshold,
-};
-
-type ContextType = keyof typeof componentMap;
-
 interface ContextState {
-  active: ContextType | null;
+  active: string | null;
+}
+
+interface ComponentMapItem {
+  description: string;
+  component: VueConstructor<Vue>;
 }
 
 const state: ContextState = reactive({
   active: null,
 });
 
-function toggle(active: ContextType | null) {
+const componentMap: Record<string, ComponentMapItem> = {
+  TypeThreshold: {
+    description: 'Threshold Controls',
+    component: TypeThreshold,
+  },
+};
+
+function register(item: ComponentMapItem) {
+  componentMap[item.component.name] = item;
+}
+
+function getComponents() {
+  const components: Record<string, VueConstructor<Vue>> = {};
+  Object.values(componentMap).forEach((v) => {
+    components[v.component.name] = v.component;
+  });
+  return components;
+}
+
+function toggle(active: string | null) {
```

Comment:
This use to have some Developer handy guarding based on the `ContextType` being a `keyof typeof ComponentMap` should we give this a more sane error if you use an invalid toggle?  The error isn't too bad currently but you would have to work your way back to see that you messed up the component name.  Feel free to resolve if you think this is clear enough already.
For Reference if you use an invalid component name:
![image](https://user-images.githubusercontent.com/61746913/158234330-8b87ca54-28b4-4dfb-b9d4-8a3a1a5e5030.png)